astropy/io/ascii/core.py

```diff
@@ -1184,12 +1184,14 @@ def read(self, table):
 
                 # otherwise, we raise an error only if it is still inconsistent
                 if len(str_vals) != n_cols:
-                    errmsg = ('Number of header columns ({}) inconsistent with'
-                              ' data columns ({}) at data line {}\n'
-                              'Header values: {}\n'
-                              'Data values: {}'.format(
+                    errmsg = (
+                        'Number of header columns ({}) inconsistent with'
+                        ' data columns ({}) at data line {}\n'
+                        'Header values: {}\n'
+                        'Data values: {}'.format(
                             n_cols, len(str_vals), i,
-                            [x.name for x in cols], str_vals))
+                            [x.name for x in cols], str_vals)
+                    )
 
```

Comment:
The problem was with the `format(`, call as hanging indents should add a level.  So I set back the whole block to add space for the `format` indentation.  This is just one of many possible solutions.